libcloud/compute/drivers/gce.py

```
@@ -1625,11 +1625,17 @@ def list_nodes(self, ex_zone=None):
             # The aggregated response returns a dict for each zone
             if zone is None:
                 for v in response['items'].values():
-                    zone_nodes = [self._to_node(i) for i in
-                                  v.get('instances', [])]
-                    list_nodes.extend(zone_nodes)
+                    for i in v.get('instances', []):
+                        try:
+                            list_nodes.append(self._to_node(i))
+                        except ResourceNotFoundError:
```

I think silently swallowing the exception justifies a comment :-)